### PR TITLE
Add trace and multi cq implementations/tests for WH Resnet

### PR DESCRIPTION
**Pull request by tt-aho**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/10020

### Problem description
Adding trace and multi cq versions of Resnet for WH

### What's changed
Refactored out logic for padding/sharding for first conv to allow for preprocessing needed for trace/multi cq

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/9845333911
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by boris-drazic — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Comment by shwetankTT** [comment]
@tt-aho Will these tests also checks functional correctness?

**Comment by tt-aho** [comment]
> @tt-aho Will these tests also checks functional correctness?

The perf tests (E2E + device) were not made to test correctness, but there are correctness tests for each of the variants in `models/experimental/resnet/tests/test_ttnn_resnet50_performant.py` which I added to nightly ci (device perf technically does do correctness tests since it's a script wrapping this test).

**Comment by shwetankTT** [comment]
> > @tt-aho Will these tests also checks functional correctness?
> 
> The perf tests (E2E + device) were not made to test correctness, but there are correctness tests for each of the variants in `models/experimental/resnet/tests/test_ttnn_resnet50_performant.py` which I added to nightly ci (device perf technically does do correctness tests since it's a script wrapping this test).

Got it, Thanks. I was adding the resnet test for new conv api in model test (tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet50_new.py). Is it already covered under nightly CI?

File: tests/scripts/run_performance.sh [hunk 1]
```diff
@@ -74,14 +76,14 @@ run_device_perf_models() {
...
 
-        env pytest "tests/ttnn/integration_tests/resnet/test_performance.py" -m $test_marker
-
         env pytest models/demos/resnet/tests -m $test_marker
     fi
 
     if [ "$tt_arch" == "wormhole_b0" ]; then
         env pytest models/demos/mamba/tests -m $test_marker
 
+        env WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest models/experimental/resnet/tests -m $test_marker
```
Comment (tt-rkim) [inline tests/scripts/run_performance.sh:85]: same deal here

Comment (tt-aho) [reply tests/scripts/run_performance.sh:85]: There are grayskull skips in the test. I was copying the format of the other tests that have the wh arch yaml set. Is it fine to set that arch when on GS?

**Review by TT-BrianLiu — APPROVED** [review]

File: tests/scripts/run_performance.sh (hunk 1 above)
Comment (tt-aho) [reply tests/scripts/run_performance.sh]: I took a look and seems like the env var will be ignored unless we are actually using WH, so I've updated the cmds to be outside the if wh statements

**Review by tt-rkim — APPROVED** [review]

**Pull request merged**